src/main/java/com/google/sps/utility/AuthenticationUtility.java

```diff
@@ -91,6 +99,78 @@ public static String generateAuthorizationHeader(HttpServletRequest request) {
     return "Bearer " + authCookie.getValue();
   }
 
+  /**
+   * Creates a Credential object to work with the Google API Java Client. Will handle verifying
+   * userId prior to creating credential. Be sure to do this before creating the credential
+   *
+   * @param request http request from client. Must contain idToken and accessToken
+   * @return a Google credential object that can be used to create an API service instance
+   */
+  public static Credential getGoogleCredential(HttpServletRequest request) {
+    // Return null if userId cannot be verified
+    try {
+      if (!verifyUserToken(request)) {
+        return null;
+      }
+    } catch (GeneralSecurityException | IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    // Return null if accessToken cannot be found
+    Cookie accessTokenCookie = getCookie(request, "accessToken");
+    String accessToken = "";
+    if (accessTokenCookie != null) {
```

Comment:
This might be a good place to use [the "return early" pattern](https://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html)

Reply:
Good idea - fixed!